sdk/main/src/TransactionRecordQuery.cc

```diff
@@ -77,10 +78,54 @@ TransactionRecord TransactionRecordQuery::mapResponse(const proto::Response& res
 }
 
 //-----
-std::function<grpc::Status(grpc::ClientContext*, const proto::Query&, proto::Response*)>
-TransactionRecordQuery::getGrpcMethod(const std::shared_ptr<internal::Node>& node) const
+Status TransactionRecordQuery::mapResponseStatus(const proto::Response& response) const
 {
-  return node->getGrpcQueryMethod(proto::Query::QueryCase::kTransactionGetRecord);
+  return STATUS_MAP.at(response.transactiongetrecord().header().nodetransactionprecheckcode());
+}
+
+//-----
+typename Executable<TransactionRecordQuery, proto::Query, proto::Response, TransactionRecord>::ExecutionStatus
+TransactionRecordQuery::shouldRetry(Status status, const Client& client, const proto::Response& response)
```

Comment:
This is pretty much duplicate code from `TransactionReceiptQuery`. Any way for this to be combined?

Reply:
Probably but I think it would be pretty clunky to set up. I'm not sure if it's worth it to try